libdirections-onboard/src/main/java/com/mapbox/navigation/route/onboard/MapboxOnboardRouter.kt

```diff
@@ -2,21 +2,88 @@ package com.mapbox.navigation.route.onboard
 
 import com.mapbox.annotation.navigation.module.MapboxNavigationModule
 import com.mapbox.annotation.navigation.module.MapboxNavigationModuleType
+import com.mapbox.navigation.base.logger.Logger
+import com.mapbox.navigation.base.route.RouteUrl
 import com.mapbox.navigation.base.route.Router
+import com.mapbox.navigation.base.route.model.Route
 import com.mapbox.navigation.base.route.model.RouteOptionsNavigation
 import com.mapbox.navigation.navigator.MapboxNativeNavigator
+import com.mapbox.navigation.navigator.MapboxNativeNavigatorImpl
+import com.mapbox.navigation.route.onboard.model.Config
+import com.mapbox.navigation.route.onboard.model.OfflineError
+import com.mapbox.navigation.route.onboard.model.mapToRouteConfig
+import com.mapbox.navigation.route.onboard.task.OfflineRouteRetrievalTask
+import com.mapbox.navigation.utils.exceptions.NavigationException
+import java.io.File
 
 @MapboxNavigationModule(MapboxNavigationModuleType.OnboardRouter, skipConfiguration = true)
-class MapboxOnboardRouter(private val navigator: MapboxNativeNavigator) : Router {
+class MapboxOnboardRouter : Router {
+
+    companion object {
+        private const val TILES_DIR_NAME = "tiles"
+    }
+
+    private val navigatorNative: MapboxNativeNavigator
+    private val config: Config
+    private val logger: Logger?
+
+    /**
+     * Creates an offline router which uses the specified offline path for storing and retrieving
+     * data.
+     *
+     * @param config offline config
+     */
+    constructor(config: Config, logger: Logger?) {
+        val tileDir = File(config.tilePath, TILES_DIR_NAME)
+        if (!tileDir.exists()) {
+            tileDir.mkdirs()
+        }
+
+        this.navigatorNative = MapboxNativeNavigatorImpl
+        this.config = config
+        this.logger = logger
+        MapboxNativeNavigatorImpl.configureRouter(config.mapToRouteConfig())
+    }
+
+    // Package private for testing purposes
+    internal constructor(
+        navigator: MapboxNativeNavigator,
+        config: Config
+    ) {
+        this.navigatorNative = navigator
+        this.config = config
+        this.logger = null
+    }
```

Comment:
```
class MapboxOnboardRouter(config: Config, private val logger: Logger?) : Router {

    companion object {
        private const val TILES_DIR_NAME = "tiles"
    }

    private val navigatorNative: MapboxNativeNavigator

    init {
        val tileDir = File(config.tilePath, TILES_DIR_NAME)
        if (!tileDir.exists()) {
            tileDir.mkdirs()
        }
        this.navigatorNative = MapboxNativeNavigatorImpl
        MapboxNativeNavigatorImpl.configureRouter(config.mapToRouteConfig())
    }
```

After that you can use `mockkObject` in unit tests to mock the navigator.

Reply:
Actually, below is gonna work even better:
```
```
class MapboxOnboardRouter(config: Config, private val logger: Logger?, private val navigatorNative: MapboxNativeNavigator = MapboxNativeNavigatorImpl) : Router {

    companion object {
        private const val TILES_DIR_NAME = "tiles"
    }

    init {
        val tileDir = File(config.tilePath, TILES_DIR_NAME)
        if (!tileDir.exists()) {
            tileDir.mkdirs()
        }
        navigatorNative.configureRouter(config.mapToRouteConfig())
    }
```
```

Reply:
in first case it doesn't work because cannot `mockObject(MapboxNativeNavigatorImpl)`, Unit tests cannot run native liblrary.
in second we give user sends own `MapboxNativeNavigator`

Reply:
:+1: 
In this case, please call one constructor from the other, to always have identical init flow.

Reply:
cannot do it because if I call `public` constructor from `internal` field `navigatorNative` is reset and It mustn't happen 😃